include/nil/blueprint/input_reader.hpp

```diff
@@ -487,74 +697,108 @@ namespace nil {
                                 if (private_input.size() == 0) {
-                                    error =
-                                        "got argument with [[private_input]] attribute, but private input file was not provided or is empty (use -p flag to provide file name).";
+                                    error << "got argument with [[private_input]] attribute, but private input file was not provided or is empty (use -p flag to provide file name).";
                                     return false;
                                 }
-                                error = "not enough values in the private input file.";
+                                error << "not enough values in the private input file.";
                                 return false;
                             }
-                        } else {
+                        }
+                        else {
                             if (public_input.size() <= pub_iter || !public_input[pub_iter].is_object()) {
                                 if (public_input.size() == 0) {
-                                    error =
-                                        "got argument without [[private_input]], but public input file was not provided or is empty (use -i flag to provide file name).";
+                                    error << "got argument without [[private_input]], but public input file was not provided or is empty (use -i flag to provide file name).";
                                     return false;
                                 }
-                                error = "not enough values in the public input file.";
+                                error << "not enough values in the public input file.";
                                 return false;
                             }
                         }
                     }
 
-                    const boost::json::object &current_value = has_values ?
-                        (is_private ?
-                        private_input[priv_iter].as_object() :
-                        public_input[pub_iter].as_object()) : boost::json::object();
+                    boost::json::value empty_value;
+                    boost::json::object empty_object;
+                    const boost::json::value &input_elem =
+                        has_values ?
+                            (is_private ? private_input[priv_iter].as_object() : public_input[pub_iter].as_object()) :
+                            empty_value;
+
+                    if (has_values && !input_elem.is_object()) {
+                        error << "Expected JSON object as a part of an input array, got \"" << input_elem << "\"";
+                        return false;
+                    }
+                    const boost::json::object &arg_obj = has_values ? input_elem.as_object() : empty_object;
+                    std::string signature;
+                    signature_parser sp;
+                    if (has_values) {
```

Comment:
Now we can parse signature only if `has_value`, right?

Reply:
Yes, that's correct, we don't have a JSON file at all if `!has_values`